src/recommender.py

```diff
@@ -106,76 +162,115 @@ def _send_action(self, speaker_id, action):
     return err if any
     '''
 
-    err = None
-    answer = 0;
+    if self.mock:
+      return None, 'mock_id'
 
-    # connect to database
-    db = pymysql.connect('localhost', 'root', '', 'ema')
-    cursor = db.cursor()
+    err = None
+    empathid = None
+    data = ''
+    time_received="NA"
+    response = -1.0
 
     # start time
     current_time = time.time()
     last_time = 0
     struc_current_time = time.localtime(current_time)
 
     # the time must be between 8:00am and 12:00pm
-    if current_time - last_time >= 300 and (struc_current_time[3] < 25 and struc_current_time[3] > 8):
+    if struc_current_time[3] < 25 and struc_current_time[3] > 8:
       last_time = current_time
 
-
       action = str(action)
 
-      survey_id = {'0':'19', '1':'20', '2':'21'}
-      #this should be 19 through 21
+      # this should be 19 through 21
+      survey_id = {'0': '19', '1': '20', '2': '21'}
 
-      #survey_id = str(action + 19)  # each action plus 19
+      # survey_id = str(action + 19)  # each action plus 19
       # add a dictionary
 
+      #time sending the prequestion
+      time1 = str(int(time.time()))
+      #date and time format of the time the prequestion is sent
+      time_sent = str(datetime.fromtimestamp(int(time1)))
 
       # items needed in url
-      pre_empathid = '999|' + str(int(time.time()))
-      empathid = None
+      pre_empathid = '999|' + time1
 
       phone_url = 'http://191.168.0.106:2226'
-      server_url = 'http://191.168.0.109/ema/ema.php'
+      server_url = 'http://191.168.0.107/ema/ema.php'
       androidid = 'db7d3cdb88e1a62a'
       alarm = 'true'
 
       # sending action to phone
       try:
-        #send prequestion 22
-        url = phone_url + '/?q={%22id%22:%22' + str(speaker_id) + '%22,%22c%22:%22startsurvey%22,%22suid%22:%22' + '22' + '%22,%22server%22:%22' + server_url + '%22,%22androidid%22:%22' + androidid + '%22,%22empathid%22:%22' + pre_empathid + '%22,%22alarm%22:%22' + alarm + '%22}'
-        #webbrowser.open(url)  # to open on browser
+        # send prequestion 22
+        url = phone_url + '/?q={%22id%22:%22' + str(speaker_id) + '%22,%22c%22:%22startsurvey%22,%22suid%22:%22' + '22' + '%22,%22server%22:%22' + \
```

Comment:
just a tiny suggestion.... do you find these encoded strings are actually so hard to read, like these `%22`
you want to refer [this api](https://stackoverflow.com/questions/8905864/url-encoding-in-python) to encode a readable one